### PR TITLE
Core/Scripts: simplify speech for DK races quests

### LINKED ISSUE #19536: text handling for dk race quests
**Description:**

The prisoners to be executed for the dk race quests "A Special Surprise" (ID 12739,12742-12750) all have the lines for all races in creature_text. They should have only the ones they actually use. Corresponding with this the c++-script has a switch command to choose for each race what to say. This is superfluous if creature_text is written properly.

**Current behaviour:**

Absolutely OK, this isn't a bug. Just bad programming. Only that the reference to broadcast_text is wrong for some of the npcs because they have all the same entries in creature_text, resulting in minor deviations from the original.

**Branch(es):**

CHANGEME 3.3.5

**TC rev. hash/commit:** 

173db1f5498dc7542e709a25f3a51d477b285f60

**TDB version:**  according to hash

**Operating system:** Windows 10

**Fix**

I will post a pull request for this.

Additionally I propose to delete broadcast_text entries belonging to cata and up for worgen and goblin dks. I will not include this in my pull request because I know how reluctant people here are to change these databases. Nevertheless it doesn't belong to 3.3.5

```SQL
DELETE FROM `broadcast_text` WHERE `ID` BETWEEN 49368 AND 49372;
DELETE FROM `broadcast_text` WHERE `ID` BETWEEN 49374 AND 49389;
DELETE FROM `locales_broadcast_text` WHERE `ID` BETWEEN 49368 AND 49372;
DELETE FROM `locales_broadcast_text` WHERE `ID` BETWEEN 49374 AND 49389;
```

### PR DESCRIPTION
**Pull request by TecDian**
**Changes proposed:**
-  eliminate the switch for the 10 races, use the same code for all
-  give the 10 npcs the correct texts with correct references to broadcast_text

**Target branch(es):** 3.3.5/master

- [X] 3.3.5

**Issues addressed:** Closes #19536

**Tests performed:** Built and tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

Nothing that I know of.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp [hunk 1]
```diff
@@ -729,310 +686,36 @@ class npc_a_special_surprise : public CreatureScript
...
-                                case 10:
-                                    Talk(SAY_EXEC_WAITING, player);
-                                    break;
-                                case 11:
-                                    Talk(EMOTE_DIES);
-                                    me->setDeathState(JUST_DIED);
-                                    me->SetHealth(0);
-                                    return;
-                            }
+                        case 0: Talk(SAY_EXEC_START, player); break;
```
Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp]: TC codestyle: place `break;` on its own line below the `case` lines and indent it.

File: sql/updates/world/3.3.5/2017_xx_xx_xx.sql [hunk 2]
```diff
@@ -0,0 +1,144 @@
+-- Malar Bravehorn, tauren
+DELETE FROM `creature_text` WHERE `entry`=29032;
+INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES 
+(29032, 0, 0, (SELECT `MaleText` FROM `broadcast_text` WHERE `ID`=29286), 12, 0, 100, 0, 0, 0, 29286, 0, 'special_surprise SAY_EXEC_START'),
```
Comment (joschiwald) [inline sql/updates/world/3.3.5/2017_xx_xx_xx.sql]: don't use subquerys with broadcast_text

Comment (DDuarte) [inline sql/updates/world/3.3.5/2017_xx_xx_xx.sql]: This should execute the subquery only once, what's the issue?

Comment (joschiwald) [inline sql/updates/world/3.3.5/2017_xx_xx_xx.sql]: cherry pick to master

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp [hunk 3]
```diff
@@ -729,310 +686,49 @@ class npc_a_special_surprise : public CreatureScript
...
+                            me->SetStandState(UNIT_STAND_STATE_KNEEL);
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                            break;
+                        case 10:
+                            Talk(SAY_EXEC_WAITING, player);
                             break;
+                        case 11:
+                            Talk(EMOTE_DIES);
+                            me->setDeathState(JUST_DIED);
+                            me->SetHealth(0);
```
Comment (ccrs) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp]: needed?

Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp]: That is the original script code, the only change is indentation.
If you have got a better suggestion, feel free to post it.

**Comment by tkrokli** [comment]
Thank you for this contribution, @TecDian ! Nice to see the script simplified like this. 👍

**Pull request merged**